connect/json/src/main/java/org/apache/kafka/connect/json/JsonDeserializer.java

```diff
@@ -35,17 +36,22 @@
      * Default constructor needed by Kafka
      */
     public JsonDeserializer() {
-        this(Collections.emptySet());
+        this(Collections.emptySet(), false);
     }
 
     /**
      * A constructor that additionally specifies some {@link DeserializationFeature}
      * for the deserializer
      *
      * @param deserializationFeatures the specified deserialization features
+     * @param exactDecimals {@code true} if trailing zeros on decimals should be maintained.
      */
-    JsonDeserializer(final Set<DeserializationFeature> deserializationFeatures) {
+    JsonDeserializer(
+        final Set<DeserializationFeature> deserializationFeatures,
+        final boolean exactDecimals
```

Comment:
what about exposing a `JsonNodeFactory` as an argument to make this more extensible?

Reply:
Can do, but is there any point for a package local constructor? We can change as we need to anyway, right?

Reply:
😈 I "have a friend" who sometimes puts new classes in the same package so that they can use package private methods

Reply:
lol. I don't know what you're talking about!

Changed...